src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStore.java

```diff
@@ -199,14 +196,28 @@ public URL toExternalURL(@NonNull Blob blob, @NonNull HttpMethod httpMethod) thr
         return builder.build().generatePresignedUrl(container, name, expiration, awsMethod);
     }
 
+    public boolean isConfigured(){
+        return StringUtils.isNotBlank(getContainer()) && StringUtils.isNotBlank(getPrefix());
+    }
+
     @Extension
     public static final class DescriptorImpl extends BlobStoreProviderDescriptor {
 
         @Override
         public String getDisplayName() {
             return "Amazon S3";
         }
-
     }
 
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder("S3BlobStore{");
+        sb.append("container='").append(getContainer()).append('\'');
+        sb.append(", prefix='").append(getPrefix()).append('\'');
+        sb.append(", region='").append(getRegion()).append('\'');
+        sb.append(", isDeleteBlobs='").append(isDeleteBlobs()).append('\'');
+        sb.append(", isDeleteStashes='").append(isDeleteStashes()).append('\'');
```

Comment:
BTW the JavaBeans convention would mean the property is called `deleteStashes` not `isDeleteStashes`. FWIW.